app/src/main/java/com/kodeco/android/countryinfo/ui/components/CountryInfoList.kt

```diff
@@ -1,27 +1,73 @@
 package com.kodeco.android.countryinfo.ui.components
 
+import android.widget.Toast
+import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.kodeco.android.countryinfo.flow.Flows
 import com.kodeco.android.countryinfo.models.Country
 import com.kodeco.android.countryinfo.sample.sampleCountries
 
 @Composable
 fun CountryInfoList(
     countries: List<Country>,
-    onRefreshClick: () -> Unit, // TODO: Utilize this onRefreshClick
+    onRefreshClick: () -> Unit,
 ) {
     var selectedCountry: Country? by remember { mutableStateOf(null) }
+    val tapCount by Flows.tapFlow.collectAsState()
+    val backCount by Flows.backFlow.collectAsState()
+    val refreshCount by Flows.refreshFlow.collectAsState()
+    val combineNavigationCount by Flows.combineNavigationFlow.collectAsState(initial = 0)
+    val context = LocalContext.current
 
     Column {
-        // TODO: Implement the Row composable here that contains the
-        //  the tap/back flow data and the refresh button.
+        Row(
```

Comment:
Minor, but you could move this Row to it's own composable/file. This way you can encapsulate the logic, including the Flow usages and not conflate it with the overall CountryInfoList composable.